src/Sentry/Exceptions/ExceptionProcessor.cs

```diff
@@ -0,0 +1,146 @@
+using System;
+using System.Collections.Generic;
+using Sentry.Extensibility;
+using Sentry.Protocol;
+
+namespace Sentry.Exceptions
+{
+    public abstract class ExceptionProcessor : ISentryEventExceptionProcessor
+    {
+        private readonly SentryOptions _options;
+
+        internal static readonly string ExceptionDataTagKey = "sentry:tag:";
+        internal static readonly string ExceptionDataContextKey = "sentry:context:";
+
+        protected ExceptionProcessor(SentryOptions options) => _options = options;
+
+        public void Process(Exception exception, SentryEvent sentryEvent)
+        {
+            _options.LogDebug("Running processor {0} on exception: {1}", GetType(), exception.Message);
+
+            var sentryExceptionList = new List<SentryException>();
+
+            foreach (var flattenedEx in Flatten(exception))
```

Comment:
The goal was to iterate on the exception list once once, but not sure it's worth rewriting this whole thing